### PR TITLE
TLMBLGIS-24 Merge Events API calls into one set

### PR DESCRIPTION
**Pull request by dovek**
Due: 5/12
@lsloan

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,12 +27,14 @@ If you know M-BoF, you don't need to know anything else.
...
-        - Messages: [http://localhost:18000/api/messages/](http://localhost:18000/api/messages/)
+        - Events: [http://localhost:18000/api/events/](http://localhost:18000/api/events/)
+        - Current Events: [http://localhost:18000/api/events/current/](http://localhost:18000/api/events/current/)
+        - Upcoming Events: [http://localhost:18000/api/events/upcoming/](http://localhost:18000/api/events/upcoming/)
         - Users: [http://localhost:18000/api/users/](http://localhost:18000/api/users/)
         - Current authen. user: [http://localhost:18000/api/me/](http://localhost:18000/api/me/)
     - Testing UI:
-        - BoF list: [http://localhost:18000/mbof/](http://localhost:18000/mbof/) (Note trailing slash.)
-        - BoF detail: [http://localhost:18000/mbof/1/](http://localhost:18000/mbof/1/)
+        - Events list: [http://localhost:18000/api/events/](http://localhost:18000/api/events/) (Note trailing slash.)
```
Comment (lsloan) [inline README.md:36]: What does it do if there's no trailing slash?

File: hacks_mbof/urls.py [hunk 2]
```diff
@@ -25,12 +25,11 @@
 router = routers.DefaultRouter()
 router.register(r'me', views.CurrentUserViewSet, base_name='me')
 router.register(r'events', views.EventViewSet)
-router.register(r'current_events', views.CurrentEventViewSet, base_name="current_events")
-router.register(r'upcoming_events', views.UpcomingEventViewSet, base_name="upcoming_events")
 router.register(r'users', views.UserViewSet)
 router.register(r'votes', views.VoteViewSet)
 
 urlpatterns = [
+    url(r'^api/events/(?P<type>[a-zA-Z ]*/$)', views.EventListViewSet.as_view()),
```
Comment (lsloan) [inline hacks_mbof/urls.py:32]: Nice!  So, ID-like strings of digits aren't matched or captured.  They're handled by the existing code instead?

File: README.md (hunk 1 above)
Comment (dovek) [inline README.md:36]: If there is no trailing slash it returns a 404

File: hacks_mbof/urls.py (hunk 2 above)
Comment (dovek) [inline hacks_mbof/urls.py:32]: Yes! If there is something like events/1/ that will still go to the event-detail page.

File: mbof/views.py [hunk 3]
```diff
@@ -28,25 +31,27 @@ class EventViewSet(viewsets.ModelViewSet):
...
+    API endpoint that allows users to be viewed or edited.
     """
+    serializer_class = EventSerializer
+
     def get_queryset(self):
-        current_time = timezone.now()
-        one_week_out = current_time + datetime.timedelta(days=7)
-        return Event.objects.filter(startTime__range=[current_time, one_week_out]).order_by('postingTime')
+        logger.debug('Type=' + self.kwargs['type'])
+        query_type = self.kwargs['type']
```
Comment (lsloan) [inline mbof/views.py:47]: Check whether `type` exists or has value before assigning its value to `query_type`.

File: mbof/views.py [hunk 4]
```diff
@@ -28,25 +31,27 @@ class EventViewSet(viewsets.ModelViewSet):
...
+        query_type = query_type.replace('/', '')
+        if query_type == 'current':
+            current_time = timezone.now()
```
Comment (lsloan) [inline mbof/views.py]: `current_time` is used in all cases, so it can go before the conditional.

**Pull request merged**